go.mod

```diff
@@ -13,25 +13,25 @@ require (
 	github.com/jteeuwen/go-bindata v3.0.8-0.20151023091102-a0ff2567cfb7+incompatible
 	github.com/onsi/ginkgo v4.7.0-origin.0+incompatible
 	github.com/onsi/gomega v1.33.1
-	github.com/openshift/api v0.0.0-20240702171116-4b89b3a92a17
-	github.com/openshift/client-go v0.0.0-20240510131258-f646d5f29250
-	github.com/openshift/cluster-kube-apiserver-operator v0.0.0-alpha.0.0.20240725130911-de2f2ac2fc9a
-	github.com/openshift/cluster-openshift-apiserver-operator v0.0.0-alpha.0.0.20240426164535-d26f3002e10b
-	github.com/openshift/cluster-openshift-controller-manager-operator v0.0.0-alpha.0.0.20240621133439-bcc371ccbc9c
-	github.com/openshift/library-go v0.0.0-20240711192904-190fec8c3f09
-	github.com/openshift/oc v0.0.0-alpha.0.0.20240802093816-e8fb3c0574da
-	github.com/openshift/openshift-controller-manager v0.0.0-alpha.0
+	github.com/openshift/api v0.0.0-20240912201240-0a8800162826
+	github.com/openshift/client-go v0.0.0-20240528061634-b054aa794d87
+	github.com/openshift/cluster-kube-apiserver-operator v0.0.0-alpha.0.0.20241023164229-aa95a2b0ecfe
+	github.com/openshift/cluster-openshift-apiserver-operator v0.0.0-alpha.0.0.20241024021503-2b2ba3bef928
+	github.com/openshift/cluster-openshift-controller-manager-operator v0.0.0-alpha.0.0.20240705084448-2b962c71d5b3
+	github.com/openshift/library-go v0.0.0-20241023075405-e3586cc1ab21
+	github.com/openshift/oc v0.0.0-alpha.0.0.20241023135555-dde885f94df5
+	github.com/openshift/openshift-controller-manager v0.0.0-alpha.0.0.20241104140441-f299e54abad2
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.19.1
 	github.com/sirupsen/logrus v1.9.3
 	github.com/spf13/cobra v1.8.1
-	k8s.io/api v0.30.4
-	k8s.io/apiextensions-apiserver v0.30.4
-	k8s.io/apimachinery v0.30.4
-	k8s.io/apiserver v0.30.4
-	k8s.io/cli-runtime v0.30.4
-	k8s.io/client-go v0.30.4
-	k8s.io/component-base v0.30.4
+	k8s.io/api v0.30.6
+	k8s.io/apiextensions-apiserver v0.30.6
+	k8s.io/apimachinery v0.30.6
+	k8s.io/apiserver v0.30.6
+	k8s.io/cli-runtime v0.30.6
+	k8s.io/client-go v1.5.2
```

Comment:
This doesn't make sense to be using since it is from 3 years ago: https://github.com/kubernetes/client-go/tree/v1.5.2

Reply:
This is addressed in the `replace` section of this file.
```
k8s.io/client-go => k8s.io/client-go v0.30.6
```

One of the other dependencies [requires](https://github.com/openshift/openshift-controller-manager/blob/release-4.17/go.mod#L30) `v1.5.2`, and go mod thinks that's the latest version since it is a greater version. When `go mod tidy` is run, the greatest version is put in this section. But the replace directive ensures we get the version we want.

Reply:
Thank you.